app/models/project.rb

```diff
@@ -45,27 +36,15 @@ class Project < ApplicationRecord
            through: :code_owner_projects,
            source: :user
 
-  validates :lang, inclusion: { in: langs.keys }
   validates :github_id, presence: true, uniqueness: true
 
-  before_save :associate_department
+  before_validation :set_default_language, on: :create
 
   private
 
-  DEPARTMENT_LANGS = {
-    backend: %w[ruby python nodejs].freeze,
-    frontend: %w[react vuejs].freeze,
-    mobile: %w[ios android react_native].freeze
-  }.freeze
-
-  def associate_department
-    return unless department.nil? || lang_changed?
-
-    department_name = find_department_name(lang)
-    self.department = Department.find_or_create_by!(name: department_name) if department_name
-  end
+  def set_default_language
```

Comment:
shouldn't this default be at DB level?

Reply:
A default language_id right? Yep I did not think on that! Thanks